commands/open-shell-container.ts

```diff
@@ -1,12 +1,20 @@
-import vscode = require('vscode');
+import * as vscode from 'vscode';
 import {ContainerItem, quickPickContainer} from './utils/quick-pick-container';
+import {DockerEngineType, docker} from './utils/docker-endpoint';
+
+const engineTypeShellCommands = {
+    [DockerEngineType.Linux]: "/bin/sh",
+    [DockerEngineType.Windows]: "cmd"
```

Comment:
I keep hearing rumor (I have no inside knowledge one way or another..even though I couldn't share it anyway) that Windows is making the switch themselves, but I question whether that would be accepted in general. I think as long as this was an option for the extension, though, you could default to whatever you want.